### PR TITLE
Svls 4825 support encrypted keys manual

### PR DESCRIPTION
**Pull request by alexgallotta**

### CODE REVIEW

File: bottlecap/src/secrets/decrypt.rs [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+                api_key: resolved_key,
+                ..config.clone()
+            })
+        } else {
+            Err(Error::new(std::io::ErrorKind::InvalidInput, "No API key or secret ARN found".to_string()))
+        };
+    }
+}
+
+struct AwsConfig {
```
Comment (alexgallotta) [inline bottlecap/src/secrets/decrypt.rs:39]: This might be better placed in the general Config loading

**Review by astuyve — CHANGES_REQUESTED** [review]
couple notes, otherwise it mostly looks good

**Comment by astuyve** [comment]
also need `cargo clippy --fix` and `cargo fmt`

File: bottlecap/src/bin/bottlecap/main.rs [hunk 2]
```diff
@@ -128,12 +130,9 @@ fn build_function_arn(account_id: &str, region: &str, function_name: &str) -> St
 #[allow(clippy::too_many_lines)]
 fn main() -> Result<()> {
     // First load the configuration
-    let lambda_directory = match env::var("LAMBDA_TASK_ROOT") {
-        Ok(val) => val,
-        Err(_) => "/var/task".to_string(),
-    };
-    let config = match config::get_config(Path::new(&lambda_directory)) {
-        Ok(config) => Arc::new(config),
+    let lambda_directory = env::var("LAMBDA_TASK_ROOT").unwrap_or_else(|_| "/var/task".to_string());
```
[the remark this answers is not in the document]
Comment (alexgallotta) [reply bottlecap/src/bin/bottlecap/main.rs:133]: yep it's yelling at me
```
error[E0593]: closure is expected to take 1 argument, but it takes 0 arguments
   --> src/bin/bottlecap/main.rs:133:57
    |
133 |     let lambda_directory = env::var("LAMBDA_TASK_ROOT").unwrap_or_else(|| "/var/task".to_string());
    |                                                         ^^^^^^^^^^^^^^ -- takes 0 arguments
    |                                                         |
    |                                                         expected closure that takes 1 argument
    |
help: consider changing the closure to take and ignore the expected argument
    |
133 |     let lambda_directory = env::var("LAMBDA_TASK_ROOT").unwrap_or_else(|_| "/var/task".to_string());
    |                                                                        ~~~

```

File: bottlecap/src/secrets/mod.rs [hunk 3]
```diff
@@ -0,0 +1 @@
+pub mod decrypt;
```
[the remark this answers is not in the document]
Comment (alexgallotta) [reply bottlecap/src/secrets/mod.rs:1]: Fixing it.
Do we have a linter/formatter for this kind of stuff?

**Comment by alexgallotta** [comment]
clippy signaled
```
warning: `bottlecap` (lib) generated 11 warnings (10 duplicates)
```
but most of them are not into this PR. Should I ignore them?

File: scripts/bottlecap/deploy_steps.sh [hunk 4]
```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+set -e
+
+ARCH="amd64"
```
[the remark this answers is not in the document]
Comment (alexgallotta) [reply scripts/bottlecap/deploy_steps.sh:4]: if it's ok, I would merge this PR and open another one with this changes

**Review by astuyve — APPROVED** [review]

**Pull request merged**